src/SimpleWeightedGraphs/simpleweightededge.jl

```diff
@@ -0,0 +1,26 @@
+"""
+$(TYPEDEF)
+
+A type representing a weighted directed edge.
+
+# Fields
+
+$(TYPEDFIELDS)
+"""
+struct SimpleWeightedEdge{T<:Integer,W} <: AbstractEdge{T,W}
+    "Source of the edge"
+    src::T
+    "Destination of the edge"
+    dst::T
+    "Weight of the edge"
+    weight::W
+end
+
+Base.Tuple(e::SimpleWeightedEdge) = (e.src, e.dst, e.weight)
+Base.isless(e1::SimpleWeightedEdge, e2::SimpleWeightedEdge) = isless(Tuple(e1), Tuple(e2))
+Base.:(==)(e1::SimpleWeightedEdge, e2::SimpleWeightedEdge) = Tuple(e1) == Tuple(e2)
+
+GraphsBase.src(e::SimpleWeightedEdge) = e.src
+GraphsBase.dst(e::SimpleWeightedEdge) = e.dst
+GraphsBase.weight(e::SimpleWeightedEdge) = e.weight
```

Comment:
We have examples where the weight would be determined in some potentially non-trivial way from vertex metadata, so it doesn't seem so natural to store it on an edge.

What about allowing `weight(g::AbstractGraph, e::AbstractEdge)`? It could then have a fallback `weight(e::AbstractEdge)` in the cases when it can be determined just from an edge.

A nice thing about that is a graph type could store a weight function as a field, and therefore it could be customizable for different instance of the same graph type.

Relatedly, with the current design the only way to customize the `weight` function is by defining a new edge type, which doesn't seem so practical in general (i.e. I would want to customize by graph type or graph instance).

Reply:
https://github.com/JuliaGraphs/GraphsBase.jl/issues/14

Reply:
Makes sense, thanks. I'm sure these same issues/questions have been brought up before, sorry for rehashing things. It's definitely more helpful to look at a concrete code proposal rather than longer abstract discussions, so thanks for making this PR!

Reply:
@etiennedeg deserves more credit, my PR is a simplified, structured version of his